### PR TITLE
Fix broken links in upgrade guide

### PR DESCRIPTION
**Pull request by melix**
Fixes broken links due to missing label

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 1]
```diff
@@ -175,13 +175,13 @@ If this is causing issues in cases you manually configure the module path, or us
...
-The `ValidateTaskProperties` task has been removed and replaced by the link:{javadocPath}/org/gradle/plugin/devel/tasks/ValidatePlugins.html task.
+The `ValidateTaskProperties` task has been removed and replaced by the link:{javadocPath}/org/gradle/plugin/devel/tasks/ValidatePlugins.html[ValidatePlugins] task.
 
 ==== Removal of `ImmutableFileCollection`
 
 The `ImmutableFileCollection` type has been removed.
-Use the link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#files-java.lang.Object...- factory method instead.
-A handle to the project layout can be obtained via link:{javadocPath}/org/gradle/api/Project.html#getLayout--.
+Use the link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#files-java.lang.Object...-[factory method] instead.
+A handle to the project layout can be obtained via link:{javadocPath}/org/gradle/api/Project.html#getLayout--[Project].
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc:184]:
```suggestion
A handle to the project layout can be obtained via link:{javadocPath}/org/gradle/api/Project.html#getLayout--[Project.layout].
```

I think the link text should mention `layout`.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 2]
```diff
@@ -194,7 +194,7 @@ The following deprecated constructors were removed:
 
 ==== Removal of arbitrary local cache configuration
 
-The local build cache configuration now needs to be done via the link:{javadocPath}/org/gradle/caching/configuration/BuildCacheConfiguration.html#local-org.gradle.api.Action-.
+The local build cache configuration now needs to be done via the link:{javadocPath}/org/gradle/caching/configuration/BuildCacheConfiguration.html#local-org.gradle.api.Action-[BuildCacheConfiguration].
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc:197]:
```suggestion
The local build cache configuration now needs to be done via link:{javadocPath}/org/gradle/caching/configuration/BuildCacheConfiguration.html#local-org.gradle.api.Action-[BuildCacheConfiguration.local()].
```

**Review by wolfs — COMMENTED** [review]
I left some comments. Do you have the rendered version somewhere?

**Comment by melix** [comment]
@bot-gradle test QFL

**Comment by melix** [comment]
@bot-gradle test QFL

**Pull request merged**